pkg/inspector/image-inspector.go

```diff
@@ -124,36 +238,118 @@ func (i *defaultImageInspector) Inspect() error {
 		return fmt.Errorf("Unable to connect to docker daemon: %v\n", err)
 	}
 
-	imageMetaBefore, inspectErrBefore := client.InspectImage(i.opts.Image)
-	if i.opts.PullPolicy == iiapi.PullNever && inspectErrBefore != nil {
-		return fmt.Errorf("Image %s is not available and pull-policy %s doesn't allow pulling",
-			i.opts.Image, i.opts.PullPolicy)
-	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if len(i.opts.Container) == 0 {
```

Comment:
nit: this is a very large `if/else` block. i would prefer to extract the contents into functions to make the code a bit more readable (e.g. `extractContainer` and `extractImage` methods; or something along those lines)

Reply:
@ilackarms I can't move it to the function. You missed the fact that I use [defer](https://github.com/openshift/image-inspector/pull/66/files/10875e1d0aafe6a73313b49fdc6e5244da56f67e#diff-cb14c05f76ccda787aced81d2a624e69R328):

```
defer mount.Unmount(i.opts.DstPath, unix.MNT_FORCE|unix.MNT_DETACH)
```

After completing the scan or any error, we must unmount the container rootfs. The context here is used to interrupt all the operations in the case of stopping the container. Therefore, it is created here and is passed to all functions.

Reply:
> ~~I can't move it to the function.~~

@ilackarms I'll make a wrapper which will reduce the size of the `Inspect()` and hide code related to the container filesystem.